### ISSUE TITLE
Faulty high DPI layout and mouse position calculation [raylib]

### ISSUE THREAD

**Issue by xypwn** [nicbarker/clay#116]
Layout calculations seem incorrect on my 1440p monitor with FLAG_WINDOW_HIGHDPI.

Running the "raylib-sidebar-scrolling-container" example, the layout initially looks normal:
![grafik](https://github.com/user-attachments/assets/69885f8d-2024-47b7-b091-61f3c655ed90)

After resizing, the content is sized larger than the window (this remains true until restarting the program):
![grafik](https://github.com/user-attachments/assets/871c693e-b14f-431d-be0a-63c7bd5c8da5)

When pulling the window over to my 1080p monitor, the mouse cursor is offset (see how the mouse cursor is to the right of the element clay/raylib thinks is selected):
![grafik](https://github.com/user-attachments/assets/e42282ca-1718-4381-a0db-641d0c353a81)

**Comment by nicbarker** [comment]
Interesting - just to clarify, this is raylib 5.5 on windows with high DPI?

**Comment by xypwn** [comment]
Yeah, sorry for not clarifying - OS is Windows 10, Raylib 5.5, 125% scaling for the main (1440p) monitor and 100% for the 1080p monitor.

**Comment by xypwn** [comment]
Hmm, this seems like a fundamental issue with Raylib: https://github.com/raysan5/raylib/issues/4215. I tried playing around with the code and while I managed to hack together a fix for the scaling issue, I was unable to fix the mouse position issue (the mouse position seems to be erroneously scaled on the second monitor based on the first monitor's scaling, so definitely some very weird and unintended stuff from Raylib's side).

In the mentioned issue, OP says that the HIGH DPI flag does more harm than good and is broken in many use cases. Thus, one approach would be to add High DPI support in Clay directly, which could simply be a factor scaling the elements and text (the user would have to handle font re-rendering on DPI change manually, but that should be possible with Raylib's GetWindowScaleDPI()).
